forem/forem
# First Few Unified Embed implementations (Liquid Tags)

## What type of PR is this? (check all applicable)


## Description
This is the first of many PRs implementing the [Unified LiquidTag](https://github.com/forem/forem/issues/15099) approach.

This PR covers these tags:
Asciinema ([example URL](https://asciinema.org/a/330532))
Codepen ([example URL](https://codepen.io/elisavetTriant/pen/KKvRRyE))
GithubGist ([example URL](https://gist.github.com/jeremyf/662585f5c4d22184a6ae133a71bf891a))
JSFiddle ([example URL](http://jsfiddle.net/link2twenty/v2kx9jcd))
NextTech ([example URL](https://nt.dev/s/6ba1fffbd09e))
Reddit ([example URL](https://www.reddit.com/r/Cricket/comments/qrkwol/match_thread_2nd_semifinal_australia_vs_pakistan/))
Soundcloud ([example URL](https://soundcloud.com/before-30-tv/stranger-moni-lati-lo-1))
TwitterTimeline ([example URL](https://twitter.com/FreyaHolmer/timelines/1215413954505297922))
Wikipedia ([example URL](https://en.wikipedia.org/wiki/Steve_Jobs))
YouTube ([example URL](https://www.youtube.com/embed/dQw4w9WgXcQ))

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/15585
closes https://github.com/forem/forem/issues/15586
closes https://github.com/forem/forem/issues/15587
closes https://github.com/forem/forem/issues/15588
closes https://github.com/forem/forem/issues/15589
closes https://github.com/forem/forem/issues/15590
closes https://github.com/forem/forem/issues/15591
closes https://github.com/forem/forem/issues/15592
closes https://github.com/forem/forem/issues/15593
closes https://github.com/forem/forem/issues/15595

## QA Instructions, Screenshots, Recordings

For these listed liquid tags, as a user, try creating liquid tags using this format: `{% embed <url-of-liquid-tag> %}`. Please use the test links included in Description above. The Liquid Tag should render properly. Also create liquid tags using the old method: `{% <liquid-tag-name> <url-of-liquid-tag> %}`. This should also work, as I am leaving the old implementation in place for now.

### UI accessibility concerns?
For the most part, no. Most of the changes are in the backend, and any frontend changes are minimal.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?
_Considering the project as a whole, I shall update the Liquid Tag documentation at the end. Since I am not changing the current implementation, the present documentation is still valid. Plus, I will avoid updating the docs piecemeal, and risk confusing users._
